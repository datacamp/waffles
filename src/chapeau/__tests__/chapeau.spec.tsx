import { render } from '@testing-library/react';

import { Chapeau } from '../index';

describe('Chapeau', () => {
  it('renders a paragraph containing the text', () => {
    const { container } = render(<Chapeau>Taylor Swift Anniversary</Chapeau>);

    const chapeau = container.querySelector('p');

    expect(chapeau).toBeInTheDocument();
    expect(chapeau).toHaveTextContent('Taylor Swift Anniversary');
  });

  it('renders snapshot', () => {
    const { getByText } = render(<Chapeau>Test</Chapeau>);

    const chapeau = getByText('Test');

    expect(chapeau).toBeInTheDocument();
    expect(chapeau).toMatchSnapshot();
  });

  it('renders snapshot of inverted variant', () => {
    const { getByText } = render(<Chapeau inverted>Test</Chapeau>);

    const chapeau = getByText('Test');

    expect(chapeau).toBeInTheDocument();
    expect(chapeau).toMatchSnapshot();
  });
});
