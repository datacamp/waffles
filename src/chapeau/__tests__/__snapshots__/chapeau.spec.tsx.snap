// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chapeau renders snapshot 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #05192D;
  font-size: 12px;
  font-weight: 800;
  line-height: 1.25;
  letter-spacing: 1.5px;
  text-transform: uppercase;
}

<p
  class="emotion-0"
>
  Test
</p>
`;

exports[`Chapeau renders snapshot of inverted variant 1`] = `
.emotion-0 {
  color: #05192D;
  font-family: Studio-Feixen-Sans,Arial,sans-serif;
  font-size: 14px;
  font-weight: 400;
  line-height: 1;
  margin: 0;
  padding: 0;
  color: #FF6EA9;
  font-size: 12px;
  font-weight: 800;
  line-height: 1.25;
  letter-spacing: 1.5px;
  text-transform: uppercase;
}

<p
  class="emotion-0"
>
  Test
</p>
`;
